coordinator/src/main.rs

```diff
@@ -0,0 +1,300 @@
+
+    // Load in the environment
+    debug!("Loading dotenv");
+    dotenv().ok();
+
+    // Loading the logger
+    debug!("Loading env_logger");
+    env_logger::init();
+
+    // Initialize the tracing (stolen from Leader Crate's init function)
+    // This will initialize a lot of the internal logging, however
+    // we will utilize the standard log for the persistent-leader.
+    // debug!("Initializing the `tracing` logger");
+    // init::tracing();
+
+    //------------------------------------------------------------------------
+    // Runtime
+    //------------------------------------------------------------------------
+
+    let runtime = {
+        debug!("Attempting to build paladin config for Runtime");
+        let config = build_paladin_config_from_env();
+
+        debug!("Determining if should initialize a prover state config...");
+        match &config.runtime {
+            paladin::config::Runtime::InMemory => {
+                info!("InMemory runtime, initializing a prover_state_manager");
+                let psm = psm::load_psm_from_env();
+                info!("Overwriting TableLoadStrategy to Monolothic due InMemory runtime");
+                let psm = psm.with_load_strategy(prover_state::TableLoadStrategy::Monolithic);
+                info!("Attempting to initialize the Prover State Manager.");
+                match PathBuf::from("circuits").exists() {
+                    true => {
+                        info!("Able to locate pre-generated circuits directory, still may take a while to load");
+                    }
+                    false => {
+                        warn!("Unable to locate a circuits directory with pre-generated circuits.  Generation will likely take a long time.");
+                    }
+                }
+
+                match psm.initialize() {
+                    Ok(_) => {
+                        info!("Initialized the ProverStateManager");
+                    }
+                    Err(err) => {
+                        error!("Failed to initialize the ProverStateManager: {}", err);
+                        panic!("Failed to initialize the ProverStateManager: {}", err);
+                    }
+                }
+            }
+            paladin_runtime => {
+                info!(
+                    "Not initializing prover_state_manager due to Paladin Runtime: {:?}",
+                    paladin_runtime
+                );
+            }
+        }
+        info!("Building Paladin Runtime");
+        match Runtime::from_config(&config, register()).await {
+            Ok(runtime) => {
+                info!("Created Paladin Runtime");
+                runtime
+            }
+            Err(err) => {
+                error!("Config: {:#?}", config);
+                error!("Error while constructing the runtime: {}", err);
+                panic!("Failed to build Paladin runtime from config: {}", err);
+            }
+        }
+    };
+
+    // Store it in an Arc
```

Comment:
This is a useful comment. I would not have known that this structure is backed by an Arc otherwise.

Assuming this comment is correct.